customer-terminate-service/app/src/test/java/se/inera/intyg/cts/infrastructure/persistence/JpaCertificateRepositoryTest.java

```diff
@@ -0,0 +1,56 @@
+package se.inera.intyg.cts.infrastructure.persistence;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static se.inera.intyg.cts.testutil.CertificateTestDataBuilder.certificateEntities;
+import static se.inera.intyg.cts.testutil.CertificateTestDataBuilder.certificates;
+import static se.inera.intyg.cts.testutil.TerminationTestDataBuilder.defaultTermination;
+import static se.inera.intyg.cts.testutil.TerminationTestDataBuilder.defaultTerminationEntity;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.infrastructure.persistence.entity.TerminationEntity;
+import se.inera.intyg.cts.infrastructure.persistence.repository.InMemoryCertificateEntityRepository;
+import se.inera.intyg.cts.infrastructure.persistence.repository.InMemoryTerminationEntityRepository;
+
+class JpaCertificateRepositoryTest {
+
+  private InMemoryTerminationEntityRepository inMemoryTerminationEntityRepository;
+  private InMemoryCertificateEntityRepository inMemoryCertificateEntityRepository;
+  private JpaCertificateRepository jpaCertificateRepository;
+  private Termination termination;
+  private TerminationEntity terminationEntity;
+
+  @BeforeEach
+  void setUp() {
+    inMemoryTerminationEntityRepository = new InMemoryTerminationEntityRepository();
+    inMemoryCertificateEntityRepository = new InMemoryCertificateEntityRepository();
+    jpaCertificateRepository = new JpaCertificateRepository(
+        inMemoryCertificateEntityRepository,
+        inMemoryTerminationEntityRepository);
+    termination = defaultTermination();
+    terminationEntity = defaultTerminationEntity();
+  }
+
+  @Test
+  void shallStoreCertificatesForExistingTermination() {
```

Comment:
The only difference from mocks and InMemory-classes is that instead of mocking I prepare the inMemory-classes with required data and can validate that the result is as expected. I think it is a good thing that the test doesn't have to mock the behaviours that the test is dependent on.